src/frontend/components/Chat/SharedChatComponents/InputDisplayNameOverlay/index.tsx

```diff
@@ -83,7 +83,13 @@ const messages = defineMessages({
   },
 });
 
-export const InputDisplayNameOverlay = () => {
+interface InputDisplayNameOverlayProps {
+  inline?: boolean;
```

Comment:
Instead of modifying this component to remove the close modal button, shouldn't you split this component:
- create a new component doing all the logic with the display name (all but the close modale button)
- update this component to mount the one you just create
This way, you can use your new component when you need it.

With your current approach, the component will yet needs to be wrapped in a `SetDisplayNamePRovider` context for no reason since this is used only to close the modale and I find it 'painfull' and can be confusion for no reason (and it would be easier to test).
What do you think ?

Reply:
useSetDisplayName is also used here : https://github.com/openfun/marsha/blob/master/src/frontend/components/Chat/SharedChatComponents/InputDisplayNameOverlay/index.tsx#L104

Reply:
yet it is still useless when we are not using a modale